### PR TITLE
cluster: let gc take care of removing k8s resources

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd-operator/issues/615

### CODE REVIEW

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -365,16 +365,15 @@ func (c *Cluster) Update(e *spec.EtcdCluster) {
...
-
 	if c.bm != nil {
-		err := c.bm.cleanup()
-		if err != nil {
+		if err := c.bm.cleanup(); err != nil {
 			c.logger.Errorf("cluster deletion: backup manager failed to cleanup: %v", err)
 		}
 	}
+
+	if err := c.gc.CollectCluster(c.cluster.Name, garbagecollection.NullUID); err != nil {
```
Comment (hongchaodeng) [inline pkg/cluster/cluster.go]: Probably not move this down here.
We should delete backup RS first before deleting pvc

Comment (xiang90) [inline pkg/cluster/cluster.go]: I moved this back. But still cluster.Delete looks strange now. I might want to not use gc.collect to remove k8s related resources in the future. Reusing that little code makes the code read worse.

File: pkg/cluster/backup_manager.go [hunk 2]
```diff
@@ -132,16 +132,10 @@ func (bm *backupManager) createBackupService() error {
 }
 
 func (bm *backupManager) cleanup() error {
-	cl, c := bm.cluster, bm.config
-	err := k8sutil.DeleteBackupReplicaSetAndService(c.KubeCli, cl.Name, cl.Namespace)
```
Comment (hongchaodeng) [inline pkg/cluster/backup_manager.go]: Also delete DeleteBackupReplicaSetAndService() method in k8sutil.go?

**Comment by hongchaodeng** [comment]
change waitBackupDeleted() in e2e/util.go to also check deletionTimestamp. Just like backup pod:
https://github.com/coreos/etcd-operator/blob/master/test/e2e/util.go#L292-L308

**Comment by hongchaodeng** [comment]
I think we also need to change gc.go to delete rs with orphanChild option to be `false`:

https://github.com/coreos/etcd-operator/blob/master/pkg/util/k8sutil/backup.go#L257-L262

Setting grace period to 0 will still leave 30s gp for backup pod though.

**Comment by xiang90** [comment]
@etcd-bot retest this please.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**